### PR TITLE
Invite email

### PR DESCRIPTION
**Pull request by ssoroka**
## Checklist

- [ ] Wrote appropriate unit tests
- [x] Considered security implications of the change
- [ ] Updated associated docs where necessary
- [x] Updated associated configuration where necessary
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

Resolves #

### CODE REVIEW

File: ui/pages/password-reset/index.js [hunk 1]
```diff
@@ -80,44 +50,7 @@ export default function PasswordReset() {
...
-            >
-              Set Password
-            </button>
-            {error && (
-              <p className='absolute -bottom-3.5 mx-auto w-full text-center text-2xs text-pink-400'>
-                {error}
-              </p>
-            )}
-          </form>
+          <PasswordResetForm />
```
Comment (hoyyeva) [inline ui/pages/password-reset/index.js:53]: is the component not in this PR?

File: docker-compose.dev.yaml [hunk 2]
```diff
@@ -43,6 +43,6 @@ services:
       - /var/lib/postgresql/data
     healthcheck:
       test: ["CMD-SHELL", "pg_isready"]
-      interval: 10s
+      interval: 15s
```
Comment (BruceMacD) [inline docker-compose.dev.yaml:46]: This seems like a good change, but was there a specific reason you found that required this change?

File: internal/server/config.go [hunk 3]
```diff
@@ -603,8 +603,14 @@ func (s Server) loadConfig(config Config) error {
 	if err := validate.Validate(config); err != nil {
 		return err
 	}
+	org, err := data.GetOrganization(s.DB(), data.ByName(models.DefaultOrganizationName))
```
Comment (BruceMacD) [inline internal/server/config.go:606]: We got rid of the ability to set the default organization name in config, correct? Just validating.

File: internal/server/data/migrations.go [hunk 4]
```diff
@@ -445,18 +445,18 @@ CREATE UNIQUE INDEX idx_grant_srp ON grants (organization_id,subject,privilege,r
 CREATE UNIQUE INDEX idx_groups_name ON groups (organization_id,name) where (deleted_at is null);
 CREATE UNIQUE INDEX idx_identities_name ON identities (organization_id,name) where (deleted_at is null);
 CREATE UNIQUE INDEX idx_providers_name ON providers (organization_id,name) where (deleted_at is null);
-CREATE UNIQUE INDEX settings_org_id ON settings (organization_id) where deleted_at is null;
+CREATE UNIQUE INDEX IF NOT EXISTS settings_org_id ON settings (organization_id) where deleted_at is null;
```
Comment (BruceMacD) [inline internal/server/data/migrations.go:448]: Should the indexes about this also be `IF NOT EXISTS`?

File: internal/server/passwordreset.go [hunk 5]
```diff
@@ -23,16 +24,11 @@ func (a *API) RequestPasswordReset(c *gin.Context, r *api.PasswordResetRequest)
 		return nil, err
 	}
 
-	// TODO: adjust domain for orgs when here
-	link, _ := url.Parse(email.AppDomain)
-	link.Path = "/password-reset"
-	query := link.Query()
-	query.Add("token", token)
-	link.RawQuery = query.Encode()
+	org := data.MustGetOrgFromContext(c)
```
Comment (BruceMacD) [inline internal/server/passwordreset.go:27]: This isn't set in middleware?

File: ui/pages/password-reset/index.js (hunk 1 above)
Comment (ssoroka) [reply ui/pages/password-reset/index.js:53]: I missed two files, ui/pages/accept-invite/index.js and ui/components/password-reset-form.js. added them now

File: internal/server/config.go (hunk 3 above)
Comment (ssoroka) [reply internal/server/config.go:606]: I guess we did.

File: internal/server/data/migrations.go (hunk 4 above)
Comment (ssoroka) [reply internal/server/data/migrations.go:448]: not sure what you mean

File: ui/components/password-reset-form.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+          }`}
+        />
+      </div>
+      <button
+        disabled={!password}
+        className='mt-6 mb-2 rounded-lg border border-violet-300 px-4 py-3 text-2xs text-violet-100 hover:border-violet-100 disabled:pointer-events-none disabled:opacity-30'
+      >
+        Set Password
+      </button>
+      {error && (
```
Comment (hoyyeva) [inline ui/components/password-reset-form.js:70]: you can use `<ErrorMessage>` component to display the error message - so the UI looks consistent throughout the app

File: ui/components/password-reset-form.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+      if (!res.ok) {
+        throw await res.json()
+      }
+
+      await res.json()
+
+      await mutate('/api/users/self')
+      router.replace('/')
+    } catch (e) {
+      setError(e)
```
Comment (hoyyeva) [inline ui/components/password-reset-form.js:34]: are we sure the `e` is always a string?

Comment (ssoroka) [reply ui/components/password-reset-form.js:34]: no idea! This code was copied from ui/pages/login/index.js

File: internal/server/passwordreset.go (hunk 5 above)
Comment (BruceMacD) [reply internal/server/passwordreset.go:27]: That was confusing, my bad. I was thinking this was setting the org on the db, but this looks fine 👍

File: internal/server/data/migrations.go (hunk 4 above)
Comment (BruceMacD) [reply internal/server/data/migrations.go:448]: The other `CREATE UNIQUE INDEX` commands *above* this (typo now edited)

**Review by BruceMacD — APPROVED** [review]
Back-end code looks good

File: ui/pages/password-reset/index.js (hunk 1 above)
Comment (jmorganca) [reply ui/pages/password-reset/index.js:54]: I think the import is missing in this file

File: ui/components/password-reset-form.js (hunk 7 above)
Comment (jmorganca) [reply ui/components/password-reset-form.js:34]: `setError(e.message)`

**Pull request merged**